frontend/src/store/root.js

```diff
@@ -1,37 +1,54 @@
 import { get } from '../utils/api'
+import { courseMapper } from '@/utils/api/course'
 
 /* root application state */
 const state = {
   error: '',
+  loading: false,
   // cached courses
-  courses: []
+  courseMap: {}, // this allows us to update courses efficiently
+  courseList: []
 }
 
 const getters = {
-  loading: ({ auth, course, questions, reviews, subject }) =>
-    [auth.loading, course.loading, questions.loading, reviews.loading, subject.loading].reduce((acc, curr) => acc || curr, false),
-  error: ({ error }) => error
+  loading: ({ loading }) => loading,
+  error: ({ error }) => error,
+  // convert object to list
+  courses: ({ courseList }) => courseList
 }
 
 const actions = {
-  populateSearch({ commit, state }) {
+  getCourses({ commit, state }) {
     // avoid repeats of this
-    if (state.courses.length) {
+    if (state.courseList.length) {
       return
     }
+    commit('TOGGLE_LOADING', true)
     return get('/course')
-      .then(data => commit('POPULATE_SEARCH', data))
+      .then(data => commit('REFRESH_COURSES', data))
       .catch(err => console.warn(err))
-  },
-  togglePageLoad() {}
+      .finally(() => commit('TOGGLE_LOADING', false))
+  }
 }
 
 const mutations = {
-  SET_LOADING(state, bool) {
+  TOGGLE_LOADING(state, bool) {
     state.loading = bool
   },
-  POPULATE_SEARCH(state, data) {
-    state.courses = data
+  REFRESH_COURSES(state, courses) {
```

Comment:
This is for what, a review update?

Reply:
Yeah, atm the idea is to load courses once only then update whenever an individual course is loaded...